### PR TITLE
Projects as arches apps #10558

### PR DESCRIPTION
**Pull request by chrabyrd**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Issues Solved

#10558

### CODE REVIEW

**Comment by chrabyrd** [comment]
@robgaston @chiatt @jacobtylerwalls 

Automatic munging of settings files ended up not being a great thing to chase. The feature as it currently is should work for our needs and I'm happy to have philosophical talks if need be 👍

**Review by jacobtylerwalls — DISMISSED** [review]
Heads up that I tested `arches-admin createproject hello` and got a directory called `{{ project_name }}`:

```sh
jwalls@Jacobs-MacBook-Pro hello % ls
README.txt		hello			package.json		yarn.lock
__init__.py		manage.py		setup.py		{{ project_name }}
gettext.config.js	nodemon.json		tsconfig.json
jwalls@Jacobs-MacBook-Pro hello % cat \{\{\ project_name\ \}\} 
cat: {{ project_name }}: Is a directory
```

**Comment by chrabyrd** [comment]
> Heads up that I tested `arches-admin createproject hello` and got a directory called `{{ project_name }}`:
> 
> ```shell
> jwalls@Jacobs-MacBook-Pro hello % ls
> README.txt		hello			package.json		yarn.lock
> __init__.py		manage.py		setup.py		{{ project_name }}
> gettext.config.js	nodemon.json		tsconfig.json
> jwalls@Jacobs-MacBook-Pro hello % cat \{\{\ project_name\ \}\} 
> cat: {{ project_name }}: Is a directory
> ```

hmm, that's not a valid command. There's `arches-admin startproject ${}` `arches-admin startapp #{}`, and `arches-project create ${}`. I tested all locally and didn't hit the issue you described. As a sanity check would you mind re-installed arches core into your local env ( `pip install -e .` ) and trying again?

**Comment by jacobtylerwalls** [comment]
Yep, thankfully a reinstall did the trick!

File: arches/install/arches-project [hunk 1]
```diff
@@ -58,8 +58,24 @@ class ArchesCommand(TemplateCommand):
...
             options["arches_version"] = f"dev/{complete_version[0]}.{complete_version[1]}.x"
         options["arches_semantic_version"] = ".".join([str(arches.VERSION[0]), str(arches.VERSION[1]), str(arches.VERSION[2])])
+
         super(ArchesCommand, self).handle('project', project_name, target, **options)
 
+        # need to manually replace instances of {{ project_name }} in some files
+        path_to_project = os.path.join(target, project_name) if target else os.path.join(os.getcwd(), project_name)
+
+        for relative_file_path in ['.yarnrc']:  # relative to app root directory
+            file = open(os.path.join(path_to_project, relative_file_path),'r')
```
Comment (jacobtylerwalls) [inline arches/install/arches-project:68]: There's an argument this is a preexisting issue, but if I provide `--destination`, the command fails:

```py
(release) jwalls@Jacobs-MacBook-Pro ~ % mkdir targetDir
(release) jwalls@Jacobs-MacBook-Pro ~ % arches-admin startproject -d targetDir newproject
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/bin/arches-admin", line 186, in <module>
    main()
  File "/Library/Frameworks/Python.framework/Versions/3.12/bin/arches-admin", line 179, in main
    COMMANDS[args.command](args)
  File "/Library/Frameworks/Python.framework/Versions/3.12/bin/arches-admin", line 148, in command_startproject
    cmd.handle(options)
  File "/Library/Frameworks/Python.framework/Versions/3.12/bin/arches-admin", line 133, in handle
    file = open(os.path.join(path_to_project, relative_file_path),'r')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'targetDir/newproject/.yarnrc'
```

File: arches/install/arches-admin [hunk 2]
```diff
@@ -167,6 +125,21 @@ class ArchesProjectCommand(TemplateCommand):
             options["arches_version"] = f"dev/{complete_version[0]}.{complete_version[1]}.x"
 
         super(ArchesProjectCommand, self).handle('project', project_name, target, **options)
+
+        # need to manually replace instances of {{ project_name }} in some files
+        path_to_project = os.path.join(target, project_name) if target else os.path.join(os.getcwd(), project_name)
+
+        for relative_file_path in ['.yarnrc']:  # relative to app root directory
+            file = open(os.path.join(path_to_project, relative_file_path),'r')
```
Comment (jacobtylerwalls) [inline arches/install/arches-admin:133]: Sorry, the traceback I posted fails here, not at the other location

File: arches/install/arches-project (hunk 1 above)
Comment (chrabyrd) [reply arches/install/arches-project:68]: This is a good catch and a good opportunity to fix the error ++

Comment (chrabyrd) [reply arches/install/arches-project:68]: fixed in latest

**Review by jacobtylerwalls — APPROVED** [review]
Looks great

**Comment by chrabyrd** [comment]
don't merge yet, investigating adding pyproject.toml

**Review by robgaston — APPROVED** [review]

**Pull request merged**